### PR TITLE
[Merged by Bors] - chore: replace IsLocalization.eq_iff_exists' by exists_of_eq

### PR DESCRIPTION
**Pull request by alreadydone**
The other direction is a consequence of `IsLocalization.map_units`.

Also do the same for `LocalizationMap` and `IsLocalizedModule`.

This means we have one less fact to prove when constructing an `IsLocalization` (etc.) instance (thus many proofs are golfed), but once we construct it we still have access to the `eq_iff_exists` lemmas (without the prime) so the API doesn't get less powerful.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
cc original authors @101damnations @jjaassoonn and users @erdOne @Vierkantor @awainverse @xroblot @mariainesdff

**Comment by erdOne** [comment]
The docstring in `Mathlib/RingTheory/Localization/Basic.lean` probably needs some change but LGTM otherwise.
Thanks for the work! It sure will make things easier.

**Comment by alreadydone** [comment]
In fact, if you're localizing at the submonoid of nonzerodivisors (i.e. you're proving an IsFractionRing instance) or a smaller submonoid, `exists_of_eq` can also be replaced by injectivity of the algebraMap, because 1 is always in the submonoid. In concrete examples we do deal with domains or fraction rings most of the time, and many current proofs indeed take `c = 1`. I might consider adding a constructor for IsLocalization that takes injectivity as an argument, because we still need some `one_mul`/`one_smul`/`coe_one` to be able to apply injectivity even after this PR.

**Comment by alreadydone** [comment]
> The docstring in `Mathlib/RingTheory/Localization/Basic.lean` probably needs some change but LGTM otherwise.

Thanks! Docstrings updated. Do they look good (or did I miss something)?

File: Mathlib/GroupTheory/MonoidLocalization.lean
Comment (alreadydone) [reply Mathlib/GroupTheory/MonoidLocalization.lean]: Rephrased in a slightly different way.

**Comment by erdOne** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Very nice! Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors merge

**Pull request closed without merging**
